app/assets/scripts/vendor/map.js

```diff
@@ -74,4 +85,36 @@
         setupClickListener('changetype-all', []);
         setupClickListener('changetype-address', ['address']);
         setupClickListener('changetype-establishment', ['establishment']);
+
+        var mapViewButton = document.getElementById("map_view");
+        var listViewButton = document.getElementById("list_view");
+
+        mapViewButton.addEventListener('click', function() {
+            $("#map").show();
+            $("#list").hide();
+        });
+
+        listViewButton.addEventListener('click', function() {
+            $("#map").hide();
+            populateListDiv();
+            $("#list").show();
+        });
+
       }
+
+      function populateListDiv() {
+          var markersToShow = [];
+          $("#list").empty();
+          var bounds = map.getBounds();
+          for (var i = 0; i < markers.length; i++) {
+              if (bounds.contains(markers[i].getPosition())) {
+                  markersToShow.push(markers[i]);
+              }
+          }
+          var html = '<table border=1 frame=void rules=rows>';
+          $.each(markersToShow, function(i, markerToShow) {
+              html += '<tr><td><br><div style="overflow:hidden;width:100%;height:100%;position:absolute"><strong>' + markerToShow.getTitle() + '</strong><br>' + markerToShow.getLabel() + '<div style="width:50px;height:50px; text-align:right; float: right;"><img src=' + markerToShow.getIcon().url + ' style="width:50px;height:50px;"></div>' + '</div><br><br><br><br></td></tr>';
```

Comment:
Really sorry if I missed this earlier.

In general, mixing HTML in the JS files is a bad idea, but sometimes it's unavoidable. The reason being that if a tiny mistake is made in the syntax of HTML here, it'll be hard to track and follow. Therefore, it's better to avoid putting actual HTML syntax in the JS scripts.

Here's what I suggest (and what I was told to do during my internship when I first put HTML into JS):
Use `document.createElement` to dynamically create DOM elements - that way these DOM elements are assigned JS variables and you can manipulate them using normal JS/jQuery syntax. Specifically, you can use `.append` to append and `.prepend` to prepend stuff within a DOM element. Here's an example from my other project:
```tableCells = [];
      for (i = 0, len = URL_COMPONENTS.length; i < len; i++) {
        urlComponent = URL_COMPONENTS[i];
        tableCell = document.createElement('td');
        $(tableCell).addClass(makeTableClass(urlComponent));
        $(tableCell).append($(makeFormId(urlComponent, true)).val());
        tableCells.push(tableCell);
      }
      updateButton = document.createElement('button');
      $(updateButton).attr({
        'type': 'button',
        'class': 'btn btn-default updateUrl'
      });
      $(updateButton).html('Update');
      deleteButton = document.createElement('button');
      $(deleteButton).attr({
        'type': 'button',
        'class': 'btn btn-default deleteUrl'
      });
      $(deleteButton).html('Delete');
      buttonGrouper = document.createElement('div');
      $(buttonGrouper).attr({
        'class': 'btn-group',
        'role': 'group'
      });
      $(buttonGrouper).append(updateButton, deleteButton);
      actionCell = document.createElement('td');
      $(actionCell).append(buttonGrouper);
      row = document.createElement('tr');
      $(row).attr({
        'data-url-id': newUrlId
      });
      $(row).append(tableCells, actionCell);
      return $('#urlTableBody').prepend(row);
```
As you can see, I dynamically created an array of table cells (each of which is an actual HTML table cell: <td></td>). Then I created two buttons: Update Button, and Delete Button, and gave them attributes and text inside. Then I put both buttons into a <div> buttonGrouper. Then I put buttonGrouper itself into a table cell called actionCell. Finally, I put the array of table cells created in the beginning, and the actionCell into a dynamicallt created row, and put the row itself into some table that already exists in the HTML page.

Note that there is no HTML syntax in here. As you can imagine, putting actual HTML syntax there would be utter mess, so it's better to do it this way. I understand it can seem annoying when HTML is little, but trust me, in the long run you will appreciate it. And though it seems like there are more lines, it's quite intuitive, and there is less room for error, so it might actually save you time.